MySQL/Scripts/Script.sql

```diff
@@ -0,0 +1,277 @@
+	(1, 102, '00005', 36),
+	(2, 102, '00005', 6),
+	(4, 102, '00005', 1);
+
+INSERT INTO activityLog (description, editedCustomerId, dateTime)
+VALUES 
+	('nothing happened', 12, CURRENT_TIMESTAMP());
+
+-- CREATE some Views
+
+CREATE VIEW v_get_products_data AS
+	SELECT p.id, p.name, p.shortDescription, pd.weight, pd.price, pd.barcodeNumber 
+		FROM product AS p
+		JOIN productDetail AS pd
+		ON p.id = pd.id;
+
+SELECT * FROM v_get_products_data; 
+
+CREATE VIEW v_get_customers_data AS
+	SELECT c.id, cd.name, c.financialId, c.vatNumber, cd.address, cd.contactPerson 
+		FROM customer AS c 
+		JOIN customerDetail AS cd 
+		ON c.id = cd.id;
+
+SELECT * FROM v_get_customers_data; 
+
+-- CREATE  search queries
+
+DROP PROCEDURE IF EXISTS sp_get_product_info_by_part_product_name;
+
+DELIMITER $$
+CREATE PROCEDURE sp_get_product_info_by_part_product_name
+(
+   productName VARCHAR(255)
+) 
+BEGIN 
+   SELECT * FROM get_products_data 
+   WHERE name LIKE CONCAT('%', productName, '%'); 
+end$$
+DELIMITER;
+
+CALL get_product_info_by_part_product_name('hai'); 
+
+-- DROP PROCEDURE sp_get_product_info_by_part_product_name;
+
+CREATE PROCEDURE sp_get_order_info_by_order_number
+(
+   orderNumber INT
+) 
+BEGIN  
+	SELECT po.orderId, c.vatNumber, cd.name, p.name, po.productQuantity, pd.price AS 'price per unit', pd.price * po.productQuantity AS 'total price' 
+		FROM productOrder AS po
+		JOIN product AS p
+		ON p.id = po.productId 
+		JOIN productDetail AS pd
+		ON p.id = pd.id 
+		JOIN customerOrder AS o
+		ON o.id = po.orderId 
+		JOIN customer AS c
+		ON c.id = o.customerId
+		JOIN customerDetail AS cd
+		ON cd.id  = c.id
+		WHERE po.orderId = orderNumber;
+END; 
+
+CALL sp_get_order_info_by_order_number(101);
+  
+-- DROP PROCEDURE sp_get_order_info_by_orderNumber;
+
+CREATE PROCEDURE sp_get_orders_for_last_month_by_customer_name
+(
+   customerName VARCHAR(255)
+) 
+BEGIN
+	SELECT o.id AS 'ORDER id', o.totalPrice, o.orderDate, cd.name AS 'customer name'
+		FROM customerOrder AS o 
+		JOIN customerDetail AS cd
+		ON cd.id = o.customerId
+		WHERE cd.name LIKE CONCAT('%', customerName, '%') 
+		AND DATE_ADD(o.orderDate, INTERVAL 1 MONTH) > current_date();
+END;
+
+CALL sp_get_orders_for_last_month_by_customer_name('Flower');
+
+-- DROP PROCEDURE sp_get_orders_for_last_month_by_customer_name;
+
+CREATE VIEW sp_get_products_sold_last_month AS
+	SELECT p.name AS 'product.name', sum(po.productQuantity) AS 'total quantity sold last month' 
+		FROM productOrder AS po 
+		JOIN productDetail AS pd
+		ON pd.id = po.productId 
+		JOIN product AS p
+		ON p.id = pd.id 
+		JOIN customerOrder AS o
+		ON o.id = po.orderId 
+		WHERE DATE_ADD(o.orderDate, INTERVAL 1 Month) > current_date()
+		GROUP BY pd.id 
+		ORDER BY p.name;
+
+SELECT * FROM sp_get_products_sold_last_month; 
+
+-- DROP view sp_get_products_sold_last_month;
+
+-- add indices
+
+CREATE INDEX index_product
+ON productDetail (id);
+
+CREATE INDEX indexOrder
+ON productOrder (orderId, productId);
```

Comment:
I dont think you need this indexes. I think there is better columns to be indexed here.